Assets/MixedRealityToolkit.Tests/PlayModeTests/BoundsControlTests.cs

```diff
@@ -0,0 +1,314 @@
+        private BoundsControl InstantiateSceneAndDefaultBbox()
+        {
+            var cube = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            cube.transform.position = Vector3.forward * 1.5f;
+            BoundsControl bbox = cube.AddComponent<BoundsControl>();
+
+            MixedRealityPlayspace.PerformTransformation(
+            p =>
+            {
+                p.position = Vector3.zero;
+                p.LookAt(cube.transform.position);
+            });
+
+            bbox.transform.localScale *= 0.5f;
+            bbox.Active = true;
+
+            return bbox;
+        }
+        #endregion
+
+        /// <summary>
+        /// Verify that we can instantiate bounds control at runtime
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator BBoxInstantiate()
+        {
+            var bbox = InstantiateSceneAndDefaultBbox();
+            yield return null;
+            Assert.IsNotNull(bbox);
+
+            GameObject.Destroy(bbox.gameObject);
+            // Wait for a frame to give Unity a change to actually destroy the object
+            yield return null;
+        }
+
+        /// <summary>
+        /// Test that if we update the bounds of a box collider, that the corners will move correctly
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator BBoxOverride()
+        {
+            var bbox = InstantiateSceneAndDefaultBbox();
+            yield return null;
+            bbox.BoundsControlActivation = UI.Experimental.BoundsControlTypes.BoundsControlActivationType.ActivateOnStart;
+            bbox.HideElementsInInspector = false;
+            yield return null;
+
+            var newObject = new GameObject();
+            var bc = newObject.AddComponent<BoxCollider>();
+            bc.center = new Vector3(.25f, 0, 0);
+            bc.size = new Vector3(0.162f, 0.1f, 1);
+            bbox.BoundsOverride = bc;
+            yield return null;
+
+            Bounds b = GetBoundsControlRigBounds(bbox);
+
+            Debug.Assert(b.center == bc.center, $"bounds center should be {bc.center} but they are {b.center}");
+            Debug.Assert(b.size == bc.size, $"bounds size should be {bc.size} but they are {b.size}");
+
+            GameObject.Destroy(bbox.gameObject);
+            GameObject.Destroy(newObject);
+            // Wait for a frame to give Unity a change to actually destroy the object
+            yield return null;
+        }
+
+        /// <summary>
+        /// Uses near interaction to scale the bounds control by directly grabbing corner
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator ScaleViaNearInteration()
+        {
+            var bbox = InstantiateSceneAndDefaultBbox();
+            yield return null;
+            var bounds = bbox.GetComponent<BoxCollider>().bounds;
+            Debug.Assert(bounds.center == new Vector3(0, 0, 1.5f));
```

Comment:
These kind of asserts should be part of the instatiate setup

Reply:
tbh i don't think these asserts should be in there at all - that's verifying that the box transform was properly applied to the BoxCollider component and that's unity internal and shouldn't be part of this test - i remove them